### DISCUSSION TITLE
Maximum single log message size of tjahzi?

### DISCUSSION THREAD

**Discussion by Retrospector** [tkowalcz/tjahzi#59] [Q&A] [answered]
Is there a maximum size for a single message transmitted by the tjahzi log4j2 appender? 
It seems that our longer messages get split at 10240 bytes, even if we increase the `batchSize` and/or `bufferSizeMegabytes`.

**Comment by tkowalcz** [comment, marked as answer]
Hi,

yes, [internally](https://github.com/tkowalcz/tjahzi/blob/master/log4j2-appender/src/main/java/pl/tkowalcz/tjahzi/log4j2/ByteBufferDestinations.java ) it uses a 10kb buffer and will fragment larger messages. It has a comment that it should be enough for everyone ;)

I can add a parameter to configure it.

**Reply by tkowalcz** [reply]
I could make it to autosize but this can cause a DoS if there is some large message due to application error.

**Reply by Retrospector** [reply]
Hi Tomasz,
thanks a lot for answering this question. Looks like our stack traces are particularly long then :)

If you find some time to add this parameter, it would be highly appreciated!

**Reply by tkowalcz** [reply]
I have a feeling that these stack traces involve Spring ;)

I should be ale to have its ready in a day or two.

**Reply by tkowalcz** [reply]
Today I will release a new version `0.9.11` (I have some problems with maven repository right now) with this configuration enabled. Checkout this for more info and documentation links - https://github.com/tkowalcz/tjahzi/releases/tag/tjahzi-parent-0.9.11.

This version is still marked as pre release as I have not deployed it to any production environment (but there is lots of tests).

**Reply by Retrospector** [reply]
Thank you again for your work! I really appreciate it! 
Will give it a test and will provide feedback.

**Reply by Retrospector** [reply]
We tested it and can confirm that our looooong stack traces now arrive in one piece.
Thanks a lot!
